src/main/java/uk/gov/hmcts/ccd/domain/casestate/EnablingConditionSorter.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.ccd.domain.casestate;
+
+import java.util.Comparator;
+import java.util.List;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.definition.EventPostStateDefinition;
+
+@Component
+public class EnablingConditionSorter {
+
+    public void sortEventPostStates(List<EventPostStateDefinition> eventPostStates) {
+        if (eventPostStates != null) {
+            eventPostStates.sort(new PostStateComparator());
```

Comment:
beware here you are sorting the list you are receiving in input as parameter rather than creating a copy. Looking at the method invocation, this means you are changing the original list stored in the CaseEventDefinition object? 
Sometimes these type of in place sorting is cause of nasty bugs...

Reply:
created the copy before sortiing